test/vcr_clj/test/core.clj

```diff
@@ -109,6 +113,25 @@
              (:args (first (calls put-foo)))))
       (is (-> (calls put-foo) first :args first meta :arg-transformer)))))
 
+(defn test-print-handlers
+  [cls]
+  (if (isa? cls java.io.InputStream)
+    (printer/tagged-handler
+      'vcr/inputstream
+      (fn [stream]
+        (.read stream)))
+
+    (serialization/default-print-handlers cls)))
+
+(deftest cassette-data-test
+  (with-spy [java-obj]
+    (with-cassette
+      {:name :boom
+       :serialization {:print-handlers test-print-handlers}}
+      [{:var #'java-obj}]
+      (is (instance? java.io.InputStream (java-obj 3)))
+      (is (= 1 (count (calls java-obj)))))))
```

Comment:
does this test fail if the `:serialization` arg is ignored?; I'm wondering if you have to run `with-cassette` twice, so that it uses the playback functionality the second time, to meaningfully test it